### ISSUE TITLE
How to implement the GoogleNet?

### ISSUE THREAD

**Issue by shiorioxy** [BVLC/caffe#1106]
Is caffe can implement the GoogleNet, which is excited me.
The 9 Inception modules seems very difficult....

**Comment by ChenglongChen** [comment]
@yindaz I have a few questions about the COMPAT_DATA layer in your googlenet patch. Looking at the also provided convert_imageset_compat2.cpp file which doesn't resize the image to a fixed size, it seems to me that COMPAT_DATA can hold data of varying size. In that case, how to compute the mean_file as required?

**Comment by npit** [comment]
@sguada may I ask, why did you use a 32 batch size, without raising the learning rate accordingly?
Thanks

**Comment by sguada** [comment]
Actually, according to  [One weird trick for parallelizing convolutional neural networks](https://github.com/BVLC/caffe/issues/430#issuecomment-43921545) when one decrease the batch size, should decrease the learning rate.
The authors used batch size 32, so I did. Also I tried batch size 128 it was performing worse.

**Comment by ducha-aiki** [comment]
@sguada, I think @npit just missed fact that caffe divides loss by batch_size and meant that to have the same parameters update scale per image, one needs to increase or decrease learning rate in proportion of number images in batch. So this is done automatically and one can use same learning rate with any batch from this point of view.

**Comment by npit** [comment]
Thank you @sguada  @ducha-aiki !
So, as a rule of thumb, I'll just have just modify the number of iterations when using different batch sizes?

**Comment by mprat** [comment]
What is the mean file for your implementation of Googlenet? Where can I find the mean image?
